kubernetes/samples/scripts/domain-lifecycle/helper.sh

```diff
@@ -478,6 +547,24 @@ function getMinReplicas {
   eval $__result=${minReplicas}
 }
 
+#
+# Function to get maximum replica count for cluster
+# $1 - Domain resource in json format
+# $2 - Name of the cluster
+# $3 - Return value containing maximum replica count
+#
+function getMaxReplicas {
+  local domainJson=$1
+  local clusterName=$2
+  local __result=$3
+
+  eval $__result=0
+  maxReplicaCmd="(.status.clusters[] | select (.clusterName == \"${clusterName}\")) \
+    | .maximumReplicas"
+  maxReplicas=$(echo ${domainJson} | jq "${maxReplicaCmd}")
+  eval $__result=${maxReplicas}
```

Comment:
there is a problem here if $__result happens to be maxReplicas -- since 'maxReplicas' should be a local var the two will complete -- choose something more unique for the local var...

Reply:
overall, it'd probably be a nice 'code cleanup' to update every function's local vars to use an __ pattern - this would eliminate the potential for them to collide with globals...